ZenJourney/app/src/main/java/ca/uwaterloo/cs/PhotobookPage.kt

```diff
@@ -83,39 +97,91 @@ fun capitalize(s: String): String {
-            ),
-            PhotobookPhoto(
-                "19 Mon",
-                BitmapFactory.decodeResource(appState.context.resources, R.drawable.pikachu_3)
-            )
-        )
+    val coroutineScope = rememberCoroutineScope()
+    val userid = "65f6591ebe57c2026bcb2300" // Hardcoded test user for now
+    val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
+
+    LaunchedEffect(Unit) {
+        try {
+            val response = PhotoApiService.getAllUserPhotos(userid, appState.dataStore.getJwt())
+            if (response.status == HttpStatusCode.OK) {
+                val listResponse: ListResponse<PhotoResponse> = response.body()
+                println(listResponse.list.toString())
+                val photoList = listResponse.list.map { photoRes ->
+                    val date = LocalDate.parse(photoRes.uploadDate, formatter)
+                    val datestr = "${date.dayOfMonth} ${
+                        capitalize(date.dayOfWeek.toString().take(3))
+                    }"
+                    PhotobookPhoto(
+                        datestr, decodeImage(photoRes.photoBase64)
+                    )
+                }
+                appState.photos.clear()
+                appState.photos.addAll(photoList.reversed())
+                println("Done updating photolist")
+                println(photoList.toString())
+            } else {
+                println("getting photolist failed")
+                val statusResponse: StatusResponse = response.body()
+                // TODO: handle failed image creation
+                println(statusResponse.body)
+                appState.pageState.value = PageStates.HOME
+            }
+        } catch (e: Exception) {
+            // TODO: handle error
+            println("some error occured while fetching photolist")
+            println(e)
+        }
     }
 
     fun addImageToPhotoState(image: Bitmap) {
-        photoState.add(
-            0,
-            PhotobookPhoto(
-                "${currentDate.dayOfMonth} ${
-                    capitalize(currentDate.dayOfWeek.toString().take(3))
-                }", image
+        coroutineScope.launch {
+            val photoRequest =
+                PhotoRequest(userid, encodeImage(image), currentDate.format(formatter))
+            try {
+                val response = PhotoApiService.createPhoto(photoRequest, appState.dataStore.getJwt())
+                if (response.status != HttpStatusCode.Created) {
+                    val statusResponse: StatusResponse = response.body()
+                    // TODO: handle failed image creation
+                    println(statusResponse.body)
+                    appState.pageState.value = PageStates.HOME
+                }
+            } catch (e: Exception) {
+                // TODO: handle error
+                println(e.message)
+            }
+            appState.photos.add(
+                0, PhotobookPhoto(
```

Comment:
what does this 0 do

Reply:
It puts the photo at the top of the list, thereby visible to the user. If we put it at the end of the list, the user can't see off first glance that the photo was added.